src/test/java/hudson/remoting/PrefetchingTest.java

```diff
@@ -131,9 +137,10 @@ public void testGetResourceAsStream() throws Exception {
      * Validates that the resource is coming from a jar.
      */
     private void verifyResource(String v) throws IOException, InterruptedException {
-        assertTrue(v, v.startsWith("jar:file:"));
-        assertTrue(v, v.contains(dir.getPath()));
-        assertTrue(v, v.endsWith("::hello"));
+        Assert.assertThat(v, StringStartsWith.startsWith("jar:file:"));
+        // need to convert to a URI otherwise this fails on windows or if you have special chars in the path
+        Assert.assertThat(v, StringContains.containsString(dir.toURI().getPath()));
+        Assert.assertThat(v, StringEndsWith.endsWith("::hello"));
```

Comment:
Nit-pick

wouldn't

`Assert.assertThat(v, allOf(StringStartsWith.startsWith("jar:file:"),StringContains.containsString(dir.toURI().getPath()),StringEndsWith.endsWith("::hello")));`

be nicer... that way you have one assert and if failing it will report exactly how it has failed... otherwise you have to run the test three times to ensure you have fixed all the potential issues:

* oh its not got the right start... fix and run
* oh its not got the right contains... fix and run
* oh its not got the right end... fix and run

vs

* oh its not got the right start, contains and end... fix and run